internal/services/giveawayservice.go

```diff
@@ -14,32 +14,46 @@ import (
 )
```

Comment:
will be simpler if there would be no difference between unconditional and conditional giveaways